coderd/reaper/detector.go

```diff
@@ -229,28 +249,22 @@ func unhangJob(ctx context.Context, log slog.Logger, db database.Store, pub pubs
 			return xerrors.Errorf("get provisioner job: %w", err)
 		}
 
-		// Check if we should still unhang it.
-		if !job.StartedAt.Valid {
-			// This shouldn't be possible to hit because the query only selects
-			// started and not completed jobs, and a job can't be "un-started".
-			return jobIneligibleError{
-				Err: xerrors.New("job is not started"),
-			}
-		}
+		jobType, threshold := reapParamsFromJob(job)
```

Comment:
The query above may need to be duplicated into a new query `GetProvisionerJobByIDForUpdate` with a `SELECT ... FOR UPDATE SKIP LOCKED` to prevent the pending job from being claimed while we're reaping it. This wasn't an issue with the hung jobs previously because they were already claimed by a provisioner

Reply:
You'll also need to handle the `ErrNoRows` return explicitly as that'll happen when it's locked.